C#/Botkit/Microsoft.BotKit.Adapters.Slack/SlackAdapter.cs

```diff
@@ -178,24 +311,249 @@ public async void ContinueConversation(ConversationReference reference, Action<T
         /// <param name="req">A request object from Restify or Express</param>
         /// <param name="res">A response object from Restify or Express</param>
         /// <param name="logic">A bot logic function in the form `async(context) => { ... }`</param>
-        public async void ProcessActivity(object req, object res, Action<TurnContext> logic)
+        public async void ProcessActivity(dynamic req, dynamic res, BotCallbackHandler logic)
         {
+            // Create an Activity based on the incoming message from Slack.
+            // There are a few different types of event that Slack might send.
 
-        }
+            dynamic slackEvent = req.body;
 
-        public override Task<ResourceResponse[]> SendActivitiesAsync(ITurnContext turnContext, Activity[] activities, CancellationToken cancellationToken)
-        {
-            throw new NotImplementedException();
-        }
+            if (slackEvent.type == "url_verification")
+            {
+                res.status(200);
+                res.send(slackEvent.challenge);
+                return;
+            }
 
-        public override Task<ResourceResponse> UpdateActivityAsync(ITurnContext turnContext, Activity activity, CancellationToken cancellationToken)
-        {
-            throw new NotImplementedException();
-        }
+            if (!VerifySignatureAsync(req, res))
+            {
+            }
+            else if (slackEvent.payload != null)
+            { 
+                // handle interactive_message callbacks and block_actions
+                slackEvent = JsonConvert.ToString(slackEvent.payload);
+                if (options.VerificationToken != null && slackEvent.token != options.VerificationToken)
+                {
+                    Console.WriteLine("Rejected due to mismatched verificationToken: ", slackEvent);
+                    res.status(403);
+                    res.end();
+                }
+                else
+                {
+                    Activity activity = new Activity()
+                    {
+                        Timestamp = new DateTime(),
+                        ChannelId = "slack",
+                        Conversation = new ConversationAccount()
+                        {
+                            Id = slackEvent.channel.id
+                            // thread_ts = slackEvent.thread_ts,
+                            // team = slackEvent.team.id
+                        },
+                        From = new ChannelAccount()
+                        {
+                            Id = slackEvent.bot_id ? slackEvent.bot_id : slackEvent.user.id
```

Comment:
This ternary should check if `slackEvent.bot_id` is not null, as `bot_id` is not a bool type